src/main/java/org/alliancegenome/curation_api/jobs/executors/Gff3Executor.java

```diff
@@ -182,24 +208,77 @@
 		return idsAdded;
 	}
 
-	private Map<String, List<Long>> loadAssociations(BulkLoadFileHistory history, List<Gff3DTO> gffData, Map<String, List<Long>> idsAdded,
-			BackendBulkDataProvider dataProvider, String assemblyId, ProcessDisplayHelper ph) {
-		for (Gff3DTO gff3Entry : gffData) {
+	private Map<String, List<Long>> loadLocationAssociations(BulkLoadFileHistory history, List<ImmutablePair<Gff3DTO, Map<String,String>>> gffData, Map<String, List<Long>> idsAdded,
+			BackendBulkDataProvider dataProvider, String assemblyId, Map<String, String> geneIdCurieMap, ProcessDisplayHelper ph) {
+		
+		for (ImmutablePair<Gff3DTO, Map<String,String>> gff3EntryPair : gffData) {
 			try {
-				idsAdded = gff3Service.loadAssociations(history, gff3Entry, idsAdded, dataProvider, assemblyId);
+				idsAdded = gff3Service.loadLocationAssociations(history, gff3EntryPair, idsAdded, dataProvider, assemblyId, geneIdCurieMap);
+				history.incrementCompleted();
+			} catch (ObjectUpdateException e) {
+				history.incrementFailed();
+				addException(history, e.getData());		
+			} catch (Exception e) {
+				e.printStackTrace();
+				history.incrementFailed();
+				addException(history, new ObjectUpdateExceptionData(gff3EntryPair.getKey(), e.getMessage(), e.getStackTrace()));
+			}
+			updateHistory(history);
+			ph.progressProcess();
+		}
+		
+		return idsAdded;
+	}
+
+	private Map<String, List<Long>> loadParentChildAssociations(BulkLoadFileHistory history, List<ImmutablePair<Gff3DTO, Map<String,String>>> gffData, Map<String, List<Long>> idsAdded,
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.
<!-- __reviewdog__:ChAyM2UzNjI1ODMxM2VlODVmEgpjaGVja3N0eWxl -->